tests/integration/test_project_navigator.py

```diff
@@ -369,9 +366,16 @@ def test_upload_submissions_plain403(self):
         self.assertEqual(modal.title, u'ERROR')
         self.assertIn(u'An error occurred while uploading your file', modal.message)
         self.assertIn(u'Technical details: 403 error', modal.message)
-        # In case message rendering text changed to escape html, this will
-        # fail and notify of the error
-        self.assertNotRegexpMatches(modal.message, r".*<\s*p\s*/?>")
+
+    def test_upload_submissions_xss(self):
+
+        marketing_pitch = self.prepare_submission()
+
+        with self.settings(ROOT_URLCONF='tests.integration.urlconf_overrides.xss'):
+            marketing_pitch.upload_file_and_return_modal(self.image_path())
+
+        modal = ModalDialogElement(self.browser)
+        self.assertIn(u'<b>XSS Test</b>', modal.message)
```

Comment:
@kaizoku Complementary to suggestion in xss.py: ... and assert that there are no browser modal dialogs opened here.